airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -320,10 +320,11 @@ def _group_streams(
     def _is_datetime_incremental_without_partition_routing(
         self,
         declarative_stream: DeclarativeStream,
-        incremental_sync_component_definition: Mapping[str, Any],
+        incremental_sync_component_definition: Mapping[str, Any] | None,
     ) -> bool:
         return (
-            bool(incremental_sync_component_definition)
+            incremental_sync_component_definition is not None
+            and bool(incremental_sync_component_definition)
```

Comment:
This was being passed a possibly "None" value and it seemed simple enough to allow it to accept None, then just treat None explicitly falsey for the test condition.

@maxi297 - Does this look okay or should we just assert or noqa that it is not None in the caller?